services/contactService.js

```diff
@@ -0,0 +1,87 @@
+const fs = require("fs").promises;
+const { nanoid } = require("nanoid");
+const path = require("path");
+
+const contactsPath = path.join(__dirname, "../", "db", "contacts.json");
+
+async function readContactsFile() {
+  const contactsData = await fs.readFile(contactsPath, "utf-8");
+  return JSON.parse(contactsData);
+}
+
+async function writeContactsFile(data) {
+  await fs.writeFile(contactsPath, JSON.stringify(data, null, 2), "utf-8");
+}
+
+async function listContacts() {
+  try {
+    return await readContactsFile();
+  } catch (error) {
+    throw error;
+  }
+}
+
+async function getContactById(contactId) {
+  try {
+    const contactsJson = await readContactsFile();
+    return contactsJson.find((contact) => contact.id === contactId) || null;
```

Comment:
Jeśli find nic nie znajdzie to zwróci undefined jakby co :)

Reply:
Cenna uwaga :) Poprawię przy kolejnym zdaniu :D